webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/EventListDisplayer.scala

```diff
@@ -120,22 +124,40 @@ class EventListDisplayer(
       displayEvents(refreshEvents())
     }
 
+
     def getEventsInterval(jsonInterval: String): JsCmd = {
       import java.sql.Timestamp
       val format = DateTimeFormat.forPattern("YYYY-MM-dd HH:mm:ss")
 
       displayEvents(for {
         parsed   <- tryo(parse(jsonInterval)) ?~! s"Error when trying to parse '${jsonInterval}' as a JSON datastructure with fields 'start' and 'end'"
         startStr <- parsed \ "start" match {
-                      case JString(startStr) => tryo(new Timestamp(DateTime.parse(startStr, format).getMillis)) ?~! s"Bad format for start date, was execpting '${format.toString}' and got '${startStr}'"
-                      case x => Failure("Error: missing start date and time")
+                      case JString(startStr) if startStr.nonEmpty =>
+                        val date = tryo(DateTime.parse(startStr, format)) ?~! s"Error when trying to parse start date '${startStr}"
+                        date match {
+                          case Full(d) => Full(Some(new Timestamp(d.getMillis)))
+                          case eb: EmptyBox =>
+                            eb ?~! s"Invalid start date"
+                        }
+                      case _ => Full(None)
                     }
         endStr   <- parsed \ "end" match {
-                      case JString(endStr) => tryo(new Timestamp(DateTime.parse(endStr, format).getMillis)) ?~! s"Bad format for end date, was execpting '${format.toString}' and got '${endStr}'"
-                      case x => Failure("Error: missing end date and time")
+                      case JString(endStr) if endStr.nonEmpty =>
+                        val date = tryo(DateTime.parse(endStr, format)) ?~! s"Error when trying to parse end date '${endStr}"
+                        date match {
+                          case Full(d) => Full(Some(new Timestamp(d.getMillis)))
+                          case eb: EmptyBox =>
+                             eb ?~! s"Invalid end date"
+                        }
+                      case _ => Full(None)
                     }
-        (start, end) = if(startStr.before(endStr)) (startStr, endStr) else (endStr, startStr)
-        logs     <- repos.getEventLogByCriteria(Some(s"creationdate >= '${start}' and creationdate < '${end}'"), None, Some("id DESC"))
+        whereStatement = (startStr, endStr) match {
```

Comment:
Your where statement should be an Option and be None in case of both strings are None!